### ISSUE TITLE
ChannelSelector num_channels parameter

### ISSUE THREAD

**Issue by namsaraeva** [MannLabs/scPortrait#82]
This parameter is redundant. Should be replaced with something better or we should get rid of it

**Comment by sophiamaedler** [comment]
How about we get rid of it and replace it with an assertion that the selected channels are smaller than the maximum number of available channels. 

Not sure how best to include it though because the transformers are defined before they are applied to the data.

**Comment by namsaraeva** [comment]
@sophiamaedler yes, i was also thinking about assertion in the method, but as you mentioned, transforms are defined before they have seen the data. maybe we should just put something in the Dataset class where transforms are applied.

**Comment by sophiamaedler** [comment]
I added some additional documentation to this transform as well as automated checks that the num_channels parameter is set to a reasonable value. I think that should be sufficient.
